### PR TITLE
Fix xcode paths

### PR DESCRIPTION
**Pull request by zeroZshadow**
#### Description
Post process the xcode project library search paths to make sure simulator and arm libraries are separated.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,7 +15,7 @@
     - Use the `buildTargetFilter` command line argument to pass in a comma delimited list of build targets to filter for. For example, `+buildTargetFilter win,macos`.
 - Added two new GDK packages: `io.improbable.worker.sdk` and `io.improbable.worker.sdk.mobile` which contain the underlying Worker SDK packages for Windows/MacOS/Linux and Android/iOS respectively. [#894](https://github.com/spatialos/gdk-for-unity/pull/894)
 - You may now `[Require]` a `WorkerId` in MonoBehaviours. For example: `[Require] private WorkerId workerId;`. [#1016](https://github.com/spatialos/gdk-for-unity/pull/1016)
-
+- iOS builds now perform a post processing step on the XCode project to ensure x86_64 and arm libraries from the SpatialOS Worker SDK are separated.
```
Comment (jamiebrynes7) [inline CHANGELOG.md]:
```suggestion
- iOS builds now perform a post processing step on the XCode project to ensure x86_64 and arm libraries from the SpatialOS Worker SDK are separated. [#1040](https://github.com/spatialos/gdk-for-unity/pull/1040)
```

File: workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+        [PostProcessBuild]
+        public static void OnPostProcessBuild(BuildTarget buildTarget, string path)
+        {
+            if (buildTarget != BuildTarget.iOS)
+            {
+                return;
+            }
+
+            Debug.Log("PostProcessBuild");
```
Comment (jessicafalk) [inline workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs]: log statement

File: workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            {
+                var type = assembly.GetTypes().FirstOrDefault(t => t.FullName == "UnityEditor.iOS.Xcode.PBXProject");
+                if (type != null)
+                {
+                    pbxType = type;
+                    break;
+                }
+            }
+
+            //pbxType = Type.GetType("UnityEditor.iOS.Xcode.PBXProject");
```
Comment (jessicafalk) [inline workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs]: commented out code

File: workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            if (pbxType == null)
+            {
+                Debug.LogWarning("iOS Support not installed.");
```
Comment (jessicafalk) [inline workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs]: maybe also tell them what to do when that happens

File: workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                .Concat(configNames.Select(configName =>
+                    InvokeMethod<string>(xcodeObject, "BuildConfigByName", targetTestingGUID, configName)));
+
+            // Update library paths for each config
+            foreach (var configGUID in configGUIDs)
+            {
+                InvokeMethod<string>(xcodeObject, "UpdateBuildPropertyForConfig", configGUID, "LIBRARY_SEARCH_PATHS",
+                    null, new[]
+                    {
+                        "$(SRCROOT)/Libraries/io.improbable.worker.sdk.mobile/Plugins/Improbable/Core/iOS/arm",
```
Comment (jessicafalk) [inline workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs]: can we store the string `$(SRCROOT)/Libraries/io.improbable.worker.sdk.mobile/Plugins/Improbable/Core/iOS` as a field? makes it a bit easier to read

File: workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+                        "$(inherited)",
+                        "$(SRCROOT)/Libraries/io.improbable.worker.sdk.mobile/Plugins/Improbable/Core/iOS/x86_64"
+                    }, null);
+            }
+
+            // Save changes to xcode project file
+            InvokeMethod<string>(xcodeObject, "WriteToFile", projPath);
+        }
+
+        private static T InvokeStaticMethod<T>(string methodName, params object[] parameters)
```
Comment (jessicafalk) [inline workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs]: that could be factored out into something more generic, but doesn't need to happen now

File: workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs (hunk 4 above)
Comment (zeroZshadow) [reply workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs]: Replaced this with an exception as this should only show up when Unity changes their API. The exception will make sure we catch it in buildkite.

File: workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs (hunk 5 above)
Comment (zeroZshadow) [reply workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs]: I rather keep these where they are right now, and it makes their context clear.

File: workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        {
+            if (buildTarget != BuildTarget.iOS)
+            {
+                return;
+            }
+
+            var assemblies = AppDomain.CurrentDomain.GetAssemblies();
+            foreach (var assembly in assemblies)
+            {
+                var type = assembly.GetTypes().FirstOrDefault(t => t.FullName == "UnityEditor.iOS.Xcode.PBXProjectddd");
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs]: `PbxProjectddd`?

File: workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+                    "LIBRARY_SEARCH_PATHS[sdk=iphonesimulator*]",
+                    new[]
+                    {
+                        "$(inherited)",
+                        "$(SRCROOT)/Libraries/io.improbable.worker.sdk.mobile/Plugins/Improbable/Core/iOS/x86_64"
+                    }, null);
+            }
+
+            // Save changes to xcode project file
+            InvokeMethod<string>(xcodeObject, "WriteToFile", projPath);
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs]: Does this actually return a `string`?

File: workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs (hunk 7 above)
Comment (zeroZshadow) [reply workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs]: On purpose to make buildkite fail.. but its not failing :/

File: workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs (hunk 8 above)
Comment (zeroZshadow) [reply workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs]: 👍

File: workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs (hunk 5 above)
Comment (jessicafalk) [reply workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs]: don't think it would make the context less clear, but fine

**Pull request merged**